plat/common/include/uk/plat/common/memory.h

```diff
@@ -33,6 +33,108 @@
 #ifndef __PLAT_CMN_MEMORY_H__
 #define __PLAT_CMN_MEMORY_H__
 
+#include <uk/config.h>
+#include <uk/assert.h>
+#include <uk/essentials.h>
+#include <uk/arch/types.h>
+#include <uk/plat/memory.h>
+
+#include <errno.h>
+#include <string.h>
+
+struct ukplat_memregion_list {
+	/** Maximum regions in the list */
+	__u32 capacity;
+	/** Current number of regions in the list */
+	__u32 count;
+	/** Array of memory regions */
+	struct ukplat_memregion_desc mrds[];
+} __packed __align(__SIZEOF_LONG__);
+
+/**
+ * Initializes the fields of an empty memory region list.
+ *
+ * @param list
+ *   The list to initialize
+ * @param capacity
+ *   The number of memory region descriptors that the list can hold. This
+ *   number must be set according to the amount of allocated memory.
+ */
+static inline void
+ukplat_memregion_list_init(struct ukplat_memregion_list *list, __u32 capacity)
+{
+	UK_ASSERT(capacity >= 1);
+
+	list->capacity = capacity;
+	list->count    = 0;
+}
+
+/**
+ * Inserts the region into the memory region list. The list is kept in
+ * ascending order of physical base addresses. Order is only preserved among
+ * ranges that do not overlap.
+ *
+ * @param list
+ *   The memory region list to insert the range into
+ * @param mrd
+ *   The memory range to insert
+ *
+ * @return
+ *   The index of the element added on success, an negative errno otherwise
+ */
+static inline int
+ukplat_memregion_list_insert(struct ukplat_memregion_list *list,
+			     const struct ukplat_memregion_desc *mrd)
+{
+	struct ukplat_memregion_desc *p;
+	__u32 i;
```

Comment:
Just a nitpick, but you return an `__u32`, yet the function's signature indicates returning an `int`.

Reply:
I will keep the `__u32` type since `i` is used multiple times with `list->count` which is `__u32`. But I added a cast `(int)i` for the return.